android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterScreen.kt

```diff
@@ -67,35 +104,80 @@ const val REGISTER_CARD_TEST_TAG = "registerCardListTestTag"
 const val FIRST_TIME_SYNC_DIALOG = "firstTimeSyncTestTag"
 const val FAB_BUTTON_REGISTER_TEST_TAG = "fabTestTag"
 const val TOP_REGISTER_SCREEN_TEST_TAG = "topScreenTestTag"
+const val SYNC_SUCCESS_TAG = "syncSuccessTag"
+const val SYNC_ERROR_TAG = "syncErrorTag"
+const val SYNC_PROGRESS_BAR_TAG = "syncProgressBarTag"
 
 @Composable
 fun RegisterScreen(
   modifier: Modifier = Modifier,
   openDrawer: (Boolean) -> Unit,
   onEvent: (RegisterEvent) -> Unit,
+  onClick: (AppMainEvent) -> Unit,
   registerUiState: RegisterUiState,
+  appUiState: AppMainUiState? = null,
   searchText: MutableState<String>,
   currentPage: MutableState<Int>,
   pagingItems: LazyPagingItems<ResourceData>,
   navController: NavController,
   toolBarHomeNavigation: ToolBarHomeNavigation = ToolBarHomeNavigation.OPEN_DRAWER,
 ) {
   val lazyListState: LazyListState = rememberLazyListState()
+  var syncNotificationBarExpanded by remember { mutableStateOf(false) }
+  val coroutineScope = rememberCoroutineScope()
+
+  var backgroundColor by remember { mutableStateOf(SideMenuTopItemDarkColor) }
+  var showSyncBar by remember { mutableStateOf(false) }
+  var applyBackgroundColor by remember { mutableStateOf(false) }
+
+  LaunchedEffect(
+    appUiState?.isSyncCompleted,
+    appUiState?.progressPercentage,
+    registerUiState.isFirstTimeSync,
+  ) {
+    when {
+      appUiState?.isSyncCompleted == SyncStatus.SUCCEEDED &&
+        appUiState.progressPercentage == 100 -> {
+        backgroundColor = Color(0xFF1DB11B)
+        showSyncBar = true
+        applyBackgroundColor = true
+        coroutineScope.launch {
+          delay(60000L)
```

Comment:
Any reason for adding this delay?

Reply:
@ellykits 
We need the delay after the sync operation is completed (whether it succeeds or fails) to give users enough time to notice the sync result before the UI reverts to its default state

Reply:
What happens when the delay is not define?